### ISSUE TITLE
Application Panic (crashed) when using vm.deal inside vm.startBroadcast 

### ISSUE THREAD

**Issue by junkim012** [foundry-rs/foundry#6540]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (893fc9f 2023-09-12T00:28:57.627748000Z)

### Operating System

macOS (Apple Silicon)

### Describe the bug

**Summary**
- When running `vm.deal` and `vm.startBroadcast` together in a script and the account runs out of funds, the script throws `The application panicked (crashed). Message: Internal EVM errror` instead of `EvmError: Out of Fund`. 

**1. First script where an account with zero native ETH balance simply tries to deposit native ETH into the WETH contract**

- Because the account has no native ETH and is attempting to deposit, it throws Out of Fund error as expected. 

```
// Throws EvmError: Out of Fund 
contract SendEthWithoutDeal is Script {
   function run() public {
        vm.startBroadcast(90743482286830539503240959006302832933333810038750515972785732718729991261126);    
        WETH.deposit{ value: 100 ether }(); 
        vm.stopBroadcast();
    } 
}
```
<img width="751" alt="Screenshot 2023-12-08 at 12 42 06 AM" src="https://github.com/foundry-rs/foundry/assets/64379343/b94cb7fb-920c-4c74-a41d-2628054e64af">

**2. Second script where I use `vm.deal` before making the WETH deposit** 

```
// throws Application Panic Internal EVM Error  
contract SendEthWithDeal is Script {
    function run() public {
        vm.startBroadcast(90743482286830539503240959006302832933333810038750515972785732718729991261126);    
        vm.deal(0x7d577a597B2742b498Cb5Cf0C26cDCD726d39E6e, 100 ether); 
        WETH.deposit{ value: 100 ether }(); 
        vm.stopBroadcast();
    }
}
```

<img width="931" alt="Screenshot 2023-12-08 at 12 44 06 AM" src="https://github.com/foundry-rs/foundry/assets/64379343/a3a55869-d316-40e3-9646-c3a310df1400">

`vm.deal` can be called before `vm.startBroadcast` and throws the same error message. 

**Note/Question** 
- Both "scripts" succeed, except the second case throws the panic on top of the success. 
- I'm also seeing that `vm.deal` does not create a transaction even inside `vm.startBroadcast`. Does this mean `vm.deal` was never meant to be used together with any broadcasting transactions and hence the error case here.

**Comment by panukettu** [comment]
yeah there is no direct integration between cheats like deal and anvil so can't broadcast things depending on them.

**Comment by Evalir** [comment]
Closed by #6549 — anvil is not integrated with cheatcodes.
